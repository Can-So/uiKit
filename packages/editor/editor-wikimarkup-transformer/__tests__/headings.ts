import WikiMarkupTransformer from '../src';

describe('JIRA wiki markup - headings', () => {
  const testCases: [string, string][] = [
    [
      'should convert string with heading in it',
      `
This is a string.
h1. Boom! this is a heading with *bold* text in it
      `,
    ],
    [
      'should set specific style for heading level 1 inside macro',
      '{panel}h1. heading{panel}',
    ],
    [
      'should set specific style for heading level 2 inside macro',
      '{panel}h2. heading{panel}',
    ],
    [
      'should set specific style for heading level 3 inside macro',
      '{panel}h3. heading{panel}',
    ],
    [
      'should set specific style for heading level 4 inside macro',
      '{panel}h4. heading{panel}',
    ],
    [
      'should set specific style for heading level 5 inside macro',
      '{panel}h5. heading{panel}',
    ],
    [
      'should set specific style for heading level 6 inside macro',
      '{panel}h6. heading{panel}',
    ],
    ['should treat h7 as usual text inside macro', '{panel}h7. heading{panel}'],
  ];

  for (const [testCaseDescription, markup] of testCases) {
    it(testCaseDescription, () => {
      const transformer = new WikiMarkupTransformer();
      expect(transformer.parse(markup)).toMatchSnapshot();
    });
  }
});
