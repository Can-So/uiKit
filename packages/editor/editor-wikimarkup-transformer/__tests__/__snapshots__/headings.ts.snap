// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JIRA wiki markup - headings should convert string with heading in it 1`] = `
Object {
  "content": Array [
    Object {
      "content": Array [
        Object {
          "text": "This is a string.",
          "type": "text",
        },
      ],
      "type": "paragraph",
    },
    Object {
      "attrs": Object {
        "level": "1",
      },
      "content": Array [
        Object {
          "text": "Boom! this is a heading with ",
          "type": "text",
        },
        Object {
          "marks": Array [
            Object {
              "type": "strong",
            },
          ],
          "text": "bold",
          "type": "text",
        },
        Object {
          "text": " text in it",
          "type": "text",
        },
      ],
      "type": "heading",
    },
    Object {
      "content": Array [
        Object {
          "text": "      ",
          "type": "text",
        },
      ],
      "type": "paragraph",
    },
  ],
  "type": "doc",
}
`;

exports[`JIRA wiki markup - headings should set specific style for heading level 1 inside macro 1`] = `
Object {
  "content": Array [
    Object {
      "attrs": Object {
        "panelType": "info",
      },
      "content": Array [
        Object {
          "content": Array [
            Object {
              "marks": Array [
                Object {
                  "type": "strong",
                },
              ],
              "text": "HEADING",
              "type": "text",
            },
          ],
          "type": "paragraph",
        },
      ],
      "type": "panel",
    },
  ],
  "type": "doc",
}
`;

exports[`JIRA wiki markup - headings should set specific style for heading level 2 inside macro 1`] = `
Object {
  "content": Array [
    Object {
      "attrs": Object {
        "panelType": "info",
      },
      "content": Array [
        Object {
          "content": Array [
            Object {
              "marks": Array [
                Object {
                  "type": "em",
                },
                Object {
                  "type": "strong",
                },
              ],
              "text": "heading",
              "type": "text",
            },
          ],
          "type": "paragraph",
        },
      ],
      "type": "panel",
    },
  ],
  "type": "doc",
}
`;

exports[`JIRA wiki markup - headings should set specific style for heading level 3 inside macro 1`] = `
Object {
  "content": Array [
    Object {
      "attrs": Object {
        "panelType": "info",
      },
      "content": Array [
        Object {
          "content": Array [
            Object {
              "marks": Array [
                Object {
                  "type": "strong",
                },
              ],
              "text": "heading",
              "type": "text",
            },
          ],
          "type": "paragraph",
        },
      ],
      "type": "panel",
    },
  ],
  "type": "doc",
}
`;

exports[`JIRA wiki markup - headings should set specific style for heading level 4 inside macro 1`] = `
Object {
  "content": Array [
    Object {
      "attrs": Object {
        "panelType": "info",
      },
      "content": Array [
        Object {
          "content": Array [
            Object {
              "marks": Array [
                Object {
                  "type": "strong",
                },
                Object {
                  "attrs": Object {
                    "color": "#97A0AF",
                  },
                  "type": "textColor",
                },
              ],
              "text": "heading",
              "type": "text",
            },
          ],
          "type": "paragraph",
        },
      ],
      "type": "panel",
    },
  ],
  "type": "doc",
}
`;

exports[`JIRA wiki markup - headings should set specific style for heading level 5 inside macro 1`] = `
Object {
  "content": Array [
    Object {
      "attrs": Object {
        "panelType": "info",
      },
      "content": Array [
        Object {
          "content": Array [
            Object {
              "marks": Array [
                Object {
                  "type": "em",
                },
                Object {
                  "attrs": Object {
                    "color": "#97A0AF",
                  },
                  "type": "textColor",
                },
              ],
              "text": "heading",
              "type": "text",
            },
          ],
          "type": "paragraph",
        },
      ],
      "type": "panel",
    },
  ],
  "type": "doc",
}
`;

exports[`JIRA wiki markup - headings should set specific style for heading level 6 inside macro 1`] = `
Object {
  "content": Array [
    Object {
      "attrs": Object {
        "panelType": "info",
      },
      "content": Array [
        Object {
          "content": Array [
            Object {
              "marks": Array [
                Object {
                  "attrs": Object {
                    "color": "#97A0AF",
                  },
                  "type": "textColor",
                },
              ],
              "text": "heading",
              "type": "text",
            },
          ],
          "type": "paragraph",
        },
      ],
      "type": "panel",
    },
  ],
  "type": "doc",
}
`;

exports[`JIRA wiki markup - headings should treat h7 as usual text inside macro 1`] = `
Object {
  "content": Array [
    Object {
      "attrs": Object {
        "panelType": "info",
      },
      "content": Array [
        Object {
          "content": Array [
            Object {
              "text": "h7. heading",
              "type": "text",
            },
          ],
          "type": "paragraph",
        },
      ],
      "type": "panel",
    },
  ],
  "type": "doc",
}
`;
